File: code/bones/java-kotlin-scala/cloud/spring-cloud-nacos/config/apollo-config/src/main/java/com/sf/config/domain/mapper/UserMapper.java
package com.sf.config.domain.mapper;

import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import com.sf.config.entity.User;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;
import org.apache.ibatis.annotations.Select;

@Mapper
public interface UserMapper extends BaseMapper<User> {
    @Select("<script>"
            +"select * from user where name = #{name} and password = #{password}"
            +"</script>")
    public User selectByNameAndPassword(@Param("name") String name,@Param("password") String password);
}

File: code/bones/java-kotlin-scala/jktdemo/src/main/java/com/sf/jkt/k/auto/mapper/UserMapper.kt
package com.sf.jkt.k.auto.mapper;

import com.sf.jkt.k.auto.entity.User;
import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import org.apache.ibatis.annotations.Mapper
import java.util.List;
import java.util.Map;

/**
 * <p>
  *  Mapper 接口
  * </p>
 *
 * @author songfei
 * @since 2019-08-13
 */
@Mapper
interface UserMapper : BaseMapper<User>{
     fun insertEx(user: User):Int
     fun deleteEx(user: User):Int
     fun updateEx(map: Map<String, Any?>):Int
     fun selectEx(map: Map<String,Any?>):List<User>
     fun insertBatchEx(users: List<User>): Int
     fun deleteBatchEx(list: List<Long>): Int
     fun updateBatchEx(list: List<Long>): Int
     fun updateBatchDifEx(users: List<User>): Int
     fun countEx(map: Map<String, Any>): Int
}
